pkg/signature/signerverifier.go

```diff
@@ -38,14 +38,16 @@ type SignerVerifier interface {
 //
 // If privateKey is an RSA key, a RSAPKCS1v15SignerVerifier will be returned. If a
 // RSAPSSSignerVerifier is desired instead, use the LoadRSAPSSSignerVerifier() method directly.
+// If privateKey is an ed25519 key, a ED25519phSignerVerifier will be returned. If a
+// Ed25519SignerVerifier is desired instead, use the LoadEd25519SignerVerifier() method directly.
 func LoadSignerVerifier(privateKey crypto.PrivateKey, hashFunc crypto.Hash) (SignerVerifier, error) {
 	switch pk := privateKey.(type) {
 	case *rsa.PrivateKey:
 		return LoadRSAPKCS1v15SignerVerifier(pk, hashFunc)
 	case *ecdsa.PrivateKey:
 		return LoadECDSASignerVerifier(pk, hashFunc)
 	case ed25519.PrivateKey:
-		return LoadED25519SignerVerifier(pk)
+		return LoadED25519phSignerVerifier(pk)
```

Comment:
As discussed offline, we'll want to keep the default as ed25519 for signers and verifiers to avoid breaking changes across the ecosystem.

I do think this is a good opportunity to figure out the right top level method for handling the "special cases" like RSA-PSS and now ed25519ph. Given only a public or private key type for rsa or ed25519, obviously it's not possible to know which is preferred between pkcs1/pss and no-ph/ph.

Now that I'm typing this, ed25519ph suppport could be added seamlessly by assuming that specifying the hashFunc to be sha512 + a ed25519 private key, we would use the prehash variant. This still is a breaking change, but at least not a change to the API. I'd like to look across the ecosystem to see if anyone is specifying sha512 currently, but I don't think they are, or they just use crypto.Hash(0) to signal not caring about the hash func.

Reply:
From a look around with `org:sigstore language:golang /(?-i)LoadSignerVerifier/` (and similar for `LoadSigner` and `LoadVerifier`) I did not spot any use except sha256.